rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -343,6 +288,7 @@ impl PendingOperation for PendingMessage {
                     .clarify_reason(ReprepareReason::ErrorEstimatingGas)
                     .await
                     .unwrap_or(ReprepareReason::ErrorEstimatingGas);
+                self.clear_metadata();
```

Comment:
iiuc this means that if we use a cached metadata that we should no longer be using and we now fail when estimating gas, we will only actually rebuild metadata in the next retry?

Because time between retries can get pretty long, I think I'd rather have us not need to wait for a second retry just to build the correct metadata, and for us to have the logic of "if we are using cached metadata, try estimating costs, and if that fails then just rebuild metadata and try estimating the costs again" explicitly in here

Reply:
Ah that makes sense, will update it